src/user/event_service.rst

```diff
@@ -0,0 +1,141 @@
+
+During job scheduling, there are overheads at many places. If the payload is too small, the overhead
+may be a big part from the payload submitted to the output returned back to users. It's good to make
+sure the payload not too small.
+
+- **scheduling period**. The basic scheduling loop is normally similar to the script described below.
+  Normally the scheduler should sleep a period between different scheduling loops. With a sleep period,
+  it will make all system busy and not efficient. If checking the configuration of workflow/workload
+  system such as Condor, Slurm, BPS, ARC CE, PanDA, Harvester and iDDS, you will find the interval
+  period defined in different systems. By default, the interval period normally can be 30 seconds
+  or 1 minute. When the system is big with many resources and users, the interval period can be even
+  bigger. At the same time, frequently there are multiple agents or threads in one system. If one job
+  involves many systems, such as iDDS->PanDA->Harvester->ARC CE->Cluster system like Slurm, and many
+  agents/threads in one system, the total overhead can be a few minutes. ::
+
+    while not terminated:
+      schedule_jobs()
+      sleep(interval_period)
+
+- **Processing overhead**. For a payload, when it begins, it needs some steps to setup the environment.
+  When it ends, it needs some time to transfer the outputs/logs to some storages. It also takes some time.
+
+EventService
+------------
+
+How does EventService run:
+
+- In EventService, events in a file is attached to a job or multiple jobs.
+
+- When pilot starts EventService, here is how does a job process multiple events::
+
+    (pilot)
+    get_job
+    setup_job
+    while more_events:
+      get_event
+      process_event
+      update_event_status_to_panda
+    terminate_job
+    upload_logs
+
+EventService introduces a fine-grained scheduling level with bookkeepings. EventService is managed as
+part of a job. It has less scheduling overheads.
+
+- EventService is managed as part of a job. It's similart to Pilot 'PULL' job, in which the scheduling is
+  between PanDA and pilot. It skips Harvester, ARC CE and Cluster system like Slurm.
+
+- Comparing to job, EventService has one setup to setup the job, then run multiple events, and at the end
+  only transfer one log file for the job. It reduces the setup time, log transfering time and so on.
+
+
+EventService for Rubin
+----------------------
+
+EventService for ATLAS is mainly used for fine-grained simulation processing, in which it normally takes
+6~10 minutes to simulate an event. For Rubin, some jobs are too short which introduce huge overheads and
+loads in scheduling and logs transfering. EventService is one idea to reduce the oveheads and loads on
+different involved systems, without cutting the bookkeeping requirements in Rubin.
+
+How to map Rubin payloads to EventService:
+
+- During workflow submission time, define a mapping file to manage the mapping between event number and
```

Comment:
This reads like something a user might need to do.  Maybe something like:  During workflow submission time, the BPS PanDA plugin defines a...